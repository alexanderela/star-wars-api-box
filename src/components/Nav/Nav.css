.Nav {
  align-items: center;
  bottom: 0px;
  display: flex;
  height: 5rem;
  justify-content: space-around;
  padding-right: 16px;
  top: 80px;
  transform: translatex(0%);
  transition: transform 0.2s ease;
  width: 80%;
}

.nav {
  color: white;
  text-decoration: none;
}

.nav:hover {
  color: #000;
}

.cat-button {
  background: #000;
  border: 1px solid #FFF;
  border-radius: 0px 100px 100px 0px;
  color: #fff;
  cursor: pointer;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 2rem;
  font-weight: 600;
  height: 3rem;
  margin: 16px 0px;
  padding: 0px;
  padding-left: 0px;
  width: 12rem;
}

.cat-button-active {
  background-color: #363535;
  box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.cat-button-inactive {
  background-color: #000;
}

.cat-button:hover {
  border-left: gray double 15px;
  color: #000;
  cursor: pointer;
  padding:0px 16px 0px 0px;
}

.cat-button:focus {
  outline: 0;
}

.people-button:hover {
  background: #d9ffd9;
  box-shadow: #d9ffd9 0px 0px 20px 2px;
  text-shadow: 
    #0F0 0px 0px 6px, 
    #0F0 0px 0px 10px;
}

.planets-button:hover {
  background:  #ffd9d9;
  box-shadow: #ffd9d9 0px 0px 20px 2px;
  text-shadow: 
    #F00 0px 0px 6px, 
    #F00 0px 0px 10px;
}

.vehicles-button:hover {
  background: #ffffd9;
  box-shadow: #ffffd9 0px 0px 20px 2px;
  text-shadow: 
    #FF0 0px 0px 6px, 
    #FF0 0px 0px 10px;
}


@media (max-width: 960px) {
  .cat-button {
    font-size: 1.75rem;
    width: 10rem;
  }
}


@media (max-width: 860px) {
  .cat-button {
    font-size: 1.5rem;
    width: 8rem;
  }
  .favorites-count {
    width: 1rem;
  }
}

@media (max-width: 690px) {
  .cat-button {
    font-size: 1.25rem;
    width: 6rem;
  }
  .favorites-count {
    width: 0.5rem;
  }
}

@media (max-width: 530px) {
  .Nav {
    flex-direction: column;
    height: auto;
  }

  .cat-button {
    font-size: 2rem;
    font-weight: 600;
    height: 3rem;
    margin: 8px 0px;
    padding: 0px;
    padding-left: 0px;
    width: 12rem;
  }
}

@media (max-width: 405px) {
  .favorites-count {
    width: 0.5rem;
  }
}