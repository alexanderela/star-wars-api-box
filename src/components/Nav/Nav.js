import React, {Component} from 'react';
import { Route, NavLink } from 'react-router-dom';
import './Nav.css'
import FavoriteButton from '../FavoriteButton/FavoriteButton.js';
import PropTypes from 'prop-types';


class Nav extends Component {
	constructor(props) {
		super(props)
		this.state = {
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false,
      favoritesSelected: false
		}
	}

	handlePeopleClick = () => {
		if (this.state.peopleSelected === true) {
			this.setState({ peopleSelected: false })
		} else {
			this.setState({ 
				peopleSelected: true,      
				vehiclesSelected: false,
      	planetsSelected: false,
      	favoritesSelected: false 
			})	
			this.props.showPeople()
		}
	}

	handlePlanetClick = () => {
		if (this.state.planetsSelected === true) {
			this.setState({ planetsSelected: false })
		} else {
			this.setState({ 
				peopleSelected: false,      
				vehiclesSelected: false,
				planetsSelected: true,
				favoritesSelected: false 
			})	
			this.props.showPlanet()
		}
	}

	handleVehicleClick = () => {
		if (this.state.vehiclesSelected === true) {
			this.setState({ vehiclesSelected: false })
		} else {
			this.setState({ 				
				peopleSelected: false,      
				vehiclesSelected: true,
				planetsSelected: false,
				favoritesSelected: false 
			})	
			this.props.showVehicle()
		}
	}

	handleFavoritesClick = () => {
		if (this.state.favoritesSelected === true) {
			this.setState({ favoritesSelected: false })
		} else {
			this.setState({
				peopleSelected: false,
				vehiclesSelected: false,
				planetsSelected: false,
				favoritesSelected: true
			})
			this.props.showFavorites()
		}
	}

render() {
	const { showPeople, showVehicles, showPlanets, favorites, showFavorites, toggleErrorPopup } = this.props
	const { peopleSelected, planetsSelected, vehiclesSelected, favoritesSelected } = this.state

	return(
		<div className="Nav">
		    <button 
		      className={`cat-button people-button 
		      	${peopleSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="people"
		      onClick={this.handlePeopleClick}
		    >
		    	<NavLink to="/people" className="nav">People</NavLink>
		    </button>
		    <button 
		      className={`cat-button planets-button 
		      	${planetsSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="planets"
		      onClick={this.handlePlanetClick}
		    >
		    	<NavLink to="/planets" className="nav">Planets</NavLink>
		    </button>
		    <button 
		      className={`cat-button vehicles-button 
		      	${vehiclesSelected 
		      		? "cat-button-active" 
		      		: "cat-button-inactive" }`}
		      name="vehicles"
		      onClick={this.handleVehicleClick}
		    >
		    	<NavLink to="/vehicles" className="nav">Vehicles</NavLink>
		    </button>
		    <FavoriteButton 
		    	className="FavoriteButton"
		    	favorites={favorites}
		    	handleFavoritesClick={this.handleFavoritesClick}
		    	favoritesSelected={favoritesSelected}
		    />
		</div>
		)
	}
}

Nav.propTypes = {
	showPeople: PropTypes.func,
	showVehicle: PropTypes.func,
	showPlanet: PropTypes.func,
	favorites: PropTypes.oneOfType([
  PropTypes.array,
  PropTypes.number
  ]),
  toggleCategoryState: PropTypes.func,
  showFavorites: PropTypes.func,
  toggleErrorPopup: PropTypes.func
}

export default Nav;