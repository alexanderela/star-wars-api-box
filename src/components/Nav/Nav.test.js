/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';
import { shallow, mount } from 'enzyme';
import { Route, NavLink } from 'react-router-dom';
import people from '../../mockData/mockPeople.js';

describe('Nav', () => {
	let wrapper;
	let mockFunction;
	let mockEvent;

	beforeEach(() => {
    mockFunction = jest.fn()
    mockEvent = { target: true }
		wrapper = shallow(<Nav   
						showPeople={mockFunction}
            showPlanet={mockFunction}
            showVehicle={mockFunction}
            toggleCategoryState={mockFunction}
            handlePeopleClick={mockFunction}/>)
	})

	it('matches the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('activates People category on click of People button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: true,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.find('.people-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

  it('calls handlePeopleClick when People button is clicked', () => {
    const spy = spyOn(wrapper.instance(), 'handlePeopleClick')
    wrapper.instance().forceUpdate()
    wrapper.find('.people-button').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

	it('deactivates People category on 2nd click of People button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ peopleSelected: true })
		wrapper.find('.people-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('activates Planets category on click of Planets button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: true
		}
		wrapper.find('.planets-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

  it('calls handlePlanetClick when Planets button is clicked', () => {
    const spy = spyOn(wrapper.instance(), 'handlePlanetClick')
    wrapper.instance().forceUpdate()
    wrapper.find('.planets-button').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

	it('deactivates Planets category on 2nd click of Planets button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ planetsSelected: true })
		wrapper.find('.planets-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

	it('activates Vehicles category on click of Vehicles button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: true,
      planetsSelected: false
		}
		wrapper.find('.vehicles-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})

  it('calls handleVehicleClick when Vehicles button is clicked', () => {
    const spy = spyOn(wrapper.instance(), 'handleVehicleClick')
    wrapper.instance().forceUpdate()
    wrapper.find('.vehicles-button').simulate('click')
    expect(spy).toHaveBeenCalled()
  })

	it('deactivates Vehicles category on 2nd click of Vehicles button', () => {
		const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
		}
		wrapper.setState({ vehiclesSelected: true })
		wrapper.find('.vehicles-button').simulate('click')
		expect(wrapper.state()).toEqual(expectedState)
	})
})

describe('handleFavoritesClick function', () => {
  let wrapper;
  let mockFunction;
  let mockEvent;

  beforeEach(() => {
    mockFunction = jest.fn()
    mockEvent = { target: true }
    wrapper = shallow(<Nav   
            showPeople={mockFunction}
            showPlanet={mockFunction}
            showVehicle={mockFunction}
            toggleCategoryState={mockFunction}
            handlePeopleClick={mockFunction}
            showFavorites={mockFunction}/>)
  })

  it('should set state if favoritesSelected is true', () => {
    const expectedState = {
      favoritesSelected: false,
      peopleSelected: false,
      vehiclesSelected: false,
      planetsSelected: false
    }
    wrapper.setState({ favoritesSelected: true })
    wrapper.instance().handleFavoritesClick()
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should set state if favoritesSelected is false', () => {
      const expectedState = {
        favoritesSelected: true,
        peopleSelected: false,
        vehiclesSelected: false,
        planetsSelected: false
      }
      wrapper.setState({ favoritesSelected: false })
      wrapper.instance().handleFavoritesClick()
      expect(wrapper.state()).toEqual(expectedState)
    })

  it('calls showFavorites', () => {
    const spy = spyOn(wrapper.instance(), 'handleFavoritesClick')
    wrapper.instance().forceUpdate()
    wrapper.instance().handleFavoritesClick()
    expect(spy).toHaveBeenCalled()
  })
})







