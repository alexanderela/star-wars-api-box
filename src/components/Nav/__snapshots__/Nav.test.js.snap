// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    handlePeopleClick={[MockFunction]}
    showPeople={[MockFunction]}
    showPlanet={[MockFunction]}
    showVehicle={[MockFunction]}
    toggleCategoryState={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="cat-button people-button 
		      	cat-button-inactive"
          name="people"
          onClick={[Function]}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/people"
          >
            People
          </NavLink>
        </button>,
        <button
          className="cat-button planets-button 
		      	cat-button-inactive"
          name="planets"
          onClick={[Function]}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/planets"
          >
            Planets
          </NavLink>
        </button>,
        <button
          className="cat-button vehicles-button 
		      	cat-button-inactive"
          name="vehicles"
          onClick={[Function]}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/vehicles"
          >
            Vehicles
          </NavLink>
        </button>,
        <FavoriteButton
          className="FavoriteButton"
          favoritesSelected={false}
          handleFavoritesClick={[Function]}
        />,
      ],
      "className": "Nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/people"
          >
            People
          </NavLink>,
          "className": "cat-button people-button 
		      	cat-button-inactive",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "People",
            "className": "nav",
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/planets"
          >
            Planets
          </NavLink>,
          "className": "cat-button planets-button 
		      	cat-button-inactive",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Planets",
            "className": "nav",
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/vehicles"
          >
            Vehicles
          </NavLink>,
          "className": "cat-button vehicles-button 
		      	cat-button-inactive",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Vehicles",
            "className": "nav",
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "FavoriteButton",
          "favorites": undefined,
          "favoritesSelected": false,
          "handleFavoritesClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="cat-button people-button 
		      	cat-button-inactive"
            name="people"
            onClick={[Function]}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/people"
            >
              People
            </NavLink>
          </button>,
          <button
            className="cat-button planets-button 
		      	cat-button-inactive"
            name="planets"
            onClick={[Function]}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/planets"
            >
              Planets
            </NavLink>
          </button>,
          <button
            className="cat-button vehicles-button 
		      	cat-button-inactive"
            name="vehicles"
            onClick={[Function]}
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/vehicles"
            >
              Vehicles
            </NavLink>
          </button>,
          <FavoriteButton
            className="FavoriteButton"
            favoritesSelected={false}
            handleFavoritesClick={[Function]}
          />,
        ],
        "className": "Nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/people"
            >
              People
            </NavLink>,
            "className": "cat-button people-button 
		      	cat-button-inactive",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "People",
              "className": "nav",
              "to": "/people",
            },
            "ref": null,
            "rendered": "People",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/planets"
            >
              Planets
            </NavLink>,
            "className": "cat-button planets-button 
		      	cat-button-inactive",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Planets",
              "className": "nav",
              "to": "/planets",
            },
            "ref": null,
            "rendered": "Planets",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/vehicles"
            >
              Vehicles
            </NavLink>,
            "className": "cat-button vehicles-button 
		      	cat-button-inactive",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Vehicles",
              "className": "nav",
              "to": "/vehicles",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "FavoriteButton",
            "favorites": undefined,
            "favoritesSelected": false,
            "handleFavoritesClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
